src/EnergyPlus/CurveManager.cc

```diff
@@ -6415,6 +6471,439 @@ Label999: ;
 
 	}
 
+	int
+	GetCurveInterpolationMethodNum( int const CurveIndex ) // index of curve in curve array
+	{
+
+		// PURPOSE OF THIS FUNCTION:
+		// get the interpolation type integer identifier for tables
+
+		// Return value
+		int TableInterpolationMethodNum;
+
+		if ( CurveIndex > 0 ) {
+			TableInterpolationMethodNum = PerfCurve( CurveIndex ).InterpolationType;
+		} else {
+			TableInterpolationMethodNum = 0;
+		}
+
+		return TableInterpolationMethodNum;
+	}
+
+	void
+	ReadTwoVarTableDataFromFile(
+		int const CurveNum,
+		std::string & FileName,
+		int & lineNum
+	)
+	{
+
+		// PURPOSE OF THIS FUNCTION:
+		// get data from an external file used to retrieve optical properties
+
+		// METHODOLOGY EMPLOYED:
+		// Recommended by NREL to read data
+
+		// Using/Aliasing
+		using DataSystemVariables::CheckForActualFileName;
+		using DataSystemVariables::TempFullFileName;
+		using General::splitString;
+
+		int TableNum;
+		bool FileExists;
+		std::ifstream infile( FileName );
+		std::string line;
+
+		CheckForActualFileName( FileName, FileExists, TempFullFileName );
```

Comment:
@lgu1234 I found the problem with reading the file on my machine.  When run with ep-launch, CheckForActualFileName looks in several places to find the file and sets TempFullFileName.  But here, infile is already set using FileName.  So, when I run this, FileExists is true, but later on the `std::getline( infile, line )` comes back false (wrong path) and he while loop never executes and lineNum is zero.